### PR TITLE
v0.3.0

### PR DESCRIPTION
**Pull request by jneug**
Like I mentioned, I did some work on the package and added a bunch of options. Maybe I went a little overboard with the changes 😄. I'm perfectly fine if you pick and choose.

Since the codebase got fairly large I did some refactoring and added some helper functions to keep things readable. But i think the code could be improved even more. For example by having a central store for default values. Maybe have a look at my [refactoring/defaults](/jneug/fork-showybox-package/tree/refactoring/defaults) branch.

### CODE REVIEW

**Comment by Pablo-Gonzalez-Calderon** [comment]
Looks good. You won me doing some changes I was thinking to do 😅. I'll review the code searching any "special cases" ;)

File: showy.typ [hunk 1]
```diff
@@ -1,197 +1,323 @@
...
+
+  line.with(
+    start: (start, 0%),
+    end: (end, 0%)
+  )
+}
+/*
+ * Function: showy-stroke()
+ *
+ * Description: Creates a stroke ot set of strokes
```
Comment (Pablo-Gonzalez-Calderon) [inline showy.typ:84]: Little typo
```suggestion
 * Description: Creates a stroke or set of strokes
```

File: showy.typ [hunk 2]
```diff
@@ -1,197 +1,323 @@
...
+  title: "",
+  footer: "",
+
+  ..body
+) = {
+  /*
+   *  Alignment wrapper
+   */
+  let alignprops = (:)
+  for prop in ("spacing", "above", "blow") {
```
Comment (Pablo-Gonzalez-Calderon) [inline showy.typ:188]: Little typo
```suggestion
  for prop in ("spacing", "above", "below") {
```

File: showy.typ [hunk 3]
```diff
@@ -1,197 +1,323 @@
...
+    weight: "bold",
+    align: left
+  ),
+  body-style: (
+    color: black,
+    align: left
+  ),
+  footer-style: (
+    color: luma(85),
+    weight: "regular",
```
Comment (Pablo-Gonzalez-Calderon) [inline showy.typ:165]: Down in the function, you use `default: bold` for footer weight, so I don't know if you would wish to use `bold` or `regular` as default 🤔

File: showy.typ [hunk 4]
```diff
@@ -1,197 +1,323 @@
...
+                    thickness: sep.at("width", default: 1pt)
+                  )
+                )
+              ))
+            )
+        )
+      )
+    )
+
+    #if footer != "" {
```
Comment (Pablo-Gonzalez-Calderon) [inline showy.typ:296]: Add a comment indicating footer starts here
```suggestion
    /*
     * Footer of the showybox
     */
    #if footer != "" {
```

**Review by Pablo-Gonzalez-Calderon — COMMENTED** [review]
There are few changes to apply. Looks great! I'm working on a manual for the package, so, after I finish it, these changes will be published in Typst's offical package repository. Thank you!

File: showy.typ [hunk 5]
```diff
@@ -1,197 +1,323 @@
...
+      block(
+        inset: if "footer-inset" in frame {
+          frame.upper-inset
```
Comment (Pablo-Gonzalez-Calderon) [inline showy.typ:299]: Should be used `footer-inset`
```suggestion
          frame.footer-inset
```

**Review by Pablo-Gonzalez-Calderon — COMMENTED** [review]
I've just found another correction ;)

File: showy.typ [hunk 6]
```diff
@@ -1,197 +1,323 @@
...
+#let showy-inset( direction, value ) = {
+  direction = repr(direction)   // allows use of alignment values
+  if type(value) == "dictionary" {
+    if direction in value {
+      value.at(direction)
+    } else if direction in ("left", "right") and "x" in value {
+      value.x
+    } else if direction in ("top", "bottom") and "y" in value {
+      value.y
+    } else {
```
[the remark this answers is not in the document]
Comment (jneug) [reply showy.typ:36]: Didn't know that was an option. You learn something new every day. ;-)

File: showy.typ (hunk 3 above)
Comment (jneug) [reply showy.typ:167]: Ah, changed it to "regular" later on. This is the reason I think a central store for defaults would be helpful.

**Comment by jneug** [comment]
Thanks for the review.

**Comment by Pablo-Gonzalez-Calderon** [comment]
Thank you again! This changes are great 😁

**Pull request merged**